src/polyfills/node-wasi.ts

```diff
@@ -6,25 +6,44 @@ import { closeSync } from 'node:fs';
 
 async function createDevNullFDs() {
   const [stdin, stdout] = await Promise.all([open(devNull, 'r'), open(devNull, 'w')]);
+  let needsClose = true;
+  // TODO: make this check always run when bun fixes [1], so `fs.promises.open()` returns a `FileHandle` as expected.
+  // [1]: https://github.com/oven-sh/bun/issues/5918
+  let close = async () => {
+    closeSync(stdin as any);
+    closeSync(stdout as any);
+  };
```

Comment:
... And Bun [returns the wrong type](https://github.com/oven-sh/bun/issues/5918) from `fs.promises.open`, so we guard against that here.